pod_test.go

```diff
@@ -35,52 +110,103 @@ func TestPodCollector(t *testing.T) {
 	// Fixed metadata on type and help text. We prepend this to every expected
 	// output so we only have to modify a single place when doing adjustments.
 	const metadata = `
-		# HELP pod_container_restarts The number of container restarts per container.
-		# TYPE pod_container_restarts counter
+		# HELP pod_container_info Information about a container in a pod.
+		# TYPE pod_container_info gauge
+		# HELP pod_container_status_ready Describes whether the containers readiness check succeeded.
+		# TYPE pod_container_status_ready gauge
+		# HELP pod_container_status_restarts The number of container restarts per container.
+		# TYPE pod_container_status_restarts counter
+		# HELP pod_container_status_running Describes whether the container is currently in running state.
+		# TYPE pod_container_status_running gauge
+		# HELP pod_container_status_terminated Describes whether the container is currently in terminated state.
+		# TYPE pod_container_status_terminated gauge
+		# HELP pod_container_status_waiting Describes whether the container is currently in waiting state.
+		# TYPE pod_container_status_waiting gauge
+		# HELP pod_info Information about pod.
+		# TYPE pod_info gauge
+		# HELP pod_status_phase The pods current phase.
+		# TYPE pod_status_phase gauge
+		# HELP pod_status_ready Describes whether the pod is ready to serve requests.
+		# TYPE pod_status_ready gauge
 	`
 	cases := []struct {
-		pods []*api.Pod
-		want string
+		pods    []*api.Pod
+		metrics []string
+		want    string
 	}{
 		{
-			pods: []*api.Pod{
-				{
-					ObjectMeta: api.ObjectMeta{
-						Name:      "pod1",
-						Namespace: "ns1",
-					},
-					Status: api.PodStatus{
-						ContainerStatuses: []api.ContainerStatus{
-							api.ContainerStatus{
-								Name:         "container1",
-								RestartCount: 0,
-							},
-						},
-					},
-				}, {
-					ObjectMeta: api.ObjectMeta{
-						Name:      "pod2",
-						Namespace: "ns2",
-					},
-					Status: api.PodStatus{
-						ContainerStatuses: []api.ContainerStatus{
-							api.ContainerStatus{
-								Name:         "container2",
-								RestartCount: 0,
-							},
-							api.ContainerStatus{
-								Name:         "container3",
-								RestartCount: 1,
-							},
-						},
-					},
-				},
-			},
+			pods: pods,
```

Comment:
Fixed @fabxc . I prefer the decomposed version, as it shows the dependency of which information is required for which metric.